fastparse/src-2/fastparse/package.scala

```diff
@@ -102,6 +102,12 @@ package object fastparse extends fastparse.SharedPackageDefs {
     def filter(f: T => Boolean)
               (implicit ctx: P[Any]): P[T] = macro MacroImpls.filterMacro[T]
 
+    /**
+     * Used by for-comprehensions.
+     */
+    def withFilter(f: T => Boolean)
+                  (implicit ctx: P[Any]): P[T] = macro MacroImpls.filterMacro[T]
+
```

Comment:
Why not just make this call filter?  Otherwise one might need to impl twice.

Reply:
We can't, it compiles an error.